executor/slow_query.go

```diff
@@ -290,32 +422,47 @@ func (e *slowQueryRetriever) parseSlowLog(ctx context.Context, sctx sessionctx.C
 	defer close(ch)
 	for {
 		startTime := time.Now()
-		log, err := e.getBatchLog(reader, &offset, logNum)
+		var logs [][]string
+		var err error
+		if !e.extractor.Desc {
+			logs, err = e.getBatchLog(reader, &offset, logNum)
+		} else {
+			logs, err = e.getBatchLogForReversedScan(reader, &offset, logNum)
+		}
 		if err != nil {
-			e.parsedSlowLogCh <- parsedSlowLog{nil, err}
+			t := slowLogTask{}
+			t.resultCh = make(chan parsedSlowLog, 1)
+			e.taskList <- t
+			t.resultCh <- parsedSlowLog{nil, err}
 			break
 		}
-		if len(log) == 0 {
+		if len(logs) == 0 || len(logs[0]) == 0 {
 			break
 		}
 		if e.stats != nil {
 			e.stats.readFile += time.Since(startTime)
 		}
-		start := offset
-		wg.Add(1)
-		ch <- 1
-		go func() {
-			defer wg.Done()
-			result, err := e.parseLog(sctx, log, start)
-			e.parsedSlowLogCh <- parsedSlowLog{result, err}
-			<-ch
-		}()
-		offset.offset = e.fileLine
-		offset.length = 0
-		select {
-		case <-ctx.Done():
-			break
-		default:
+		for i := range logs {
+			log := logs[i]
+			t := slowLogTask{}
+			t.resultCh = make(chan parsedSlowLog, 1)
+			start := offset
+			wg.Add(1)
+			ch <- 1
+			e.taskList <- t
+			go func() {
+				defer wg.Done()
+				result, err := e.parseLog(sctx, log, start)
+				t.resultCh <- parsedSlowLog{result, err}
+				<-ch
+			}()
+			offset.offset = e.fileLine
+			offset.length = 0
+			select {
+			case <-ctx.Done():
+				break
```

Comment:
```suggestion
				return
```

Reply:
the `break` will only break out the `select` in line 461. 
we can just use return here, and no need to wait `wg.Wait` finish any more.